packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+        ...feelessVerificationState.komenci,
+        unverifiedMtwAddress: verifiedMtwAddress,
+      },
+    })
+  )
+
+  return verifiedMtwAddress
+}
+
+function* reFetchVerifiedMtw(
+  contractKit: ContractKit,
+  walletAddress: string,
+  e164Number: string,
+  startingPepperQuota: number
+) {
+  Logger.debug(TAG, '@reFetchVerifiedMtw', 'Starting fetch')
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  const { pepperQuotaRemaining } = feelessVerificationState.komenci
+  // If Komenci was used to fetch their pepper, check if the they have a verified
+  // account. If user already had their pepper, then check would have happened
+  // in `feelessFetchVerificationState`
+  if (startingPepperQuota > pepperQuotaRemaining) {
+    yield call(fetchVerifiedMtw, contractKit, walletAddress, e164Number)
+  }
+}
+
+function* isMtwVerified() {
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  return feelessVerificationState.status.isVerified
+}
+
+function* fetchAttestationStatus(contractKit: ContractKit) {
+  Logger.debug(TAG, '@fetchAttestationStatus', 'Starting fetch')
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  const { phoneHashDetails, komenci } = feelessVerificationState
+  const { phoneHash } = phoneHashDetails
+  const { unverifiedMtwAddress } = komenci
+
+  // If there isn't an address stored in state or we already know that the
+  // MTW is verified, then there is nothing to check the progress of
+  if (!unverifiedMtwAddress || feelessVerificationState.status.isVerified) {
+    return
+  }
+
+  const attestationsWrapper: AttestationsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAttestations,
+  ])
+
+  const status: AttestationsStatus = yield call(
+    getAttestationsStatus,
+    attestationsWrapper,
+    unverifiedMtwAddress,
+    phoneHash
+  )
+
+  const actionableAttestations: ActionableAttestation[] = yield call(
+    getActionableAttestations,
+    attestationsWrapper,
+    phoneHash,
+    unverifiedMtwAddress
+  )
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      actionableAttestations,
+      status,
+    })
+  )
+
+  return status
+}
+
+function* startOrResumeKomenciSession(komenciKit: KomenciKit, e164Number: string) {
+  Logger.debug(TAG, '@startOrResumeKomenciSession', 'Starting session')
+  // Fetch session state to make sure we have the most up-to-date session info
+  yield call(fetchSessionState, komenciKit, e164Number)
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  const { sessionActive, captchaToken, sessionToken } = feelessVerificationState.komenci
+
+  // If there isn't an active session, start one. Need to include `sessionActive`
```

Comment:
Because there is value in knowing the state of your session before you're ready to start a new one.

For example, we have the first `fetchKomenciSessionState` in the `feelessFetchVerificationState` generator, which happens immediately when the user lands on the "start verification screen". If there is not one currently active, then the user must complete a captcha before we can start a new one for them. Given the need for user action between when a session is known to be inactive and when we are able to start a new one, it made most sense for me to keep these two functions separate.